Makefile.am

```diff
@@ -443,6 +443,10 @@ RRD_PLUGIN_FILES = \
     database/rrdsetvar.h \
     database/rrdvar.c \
     database/rrdvar.h \
+    database/rrddim_mem.c \
+    database/rrddim_mem.h \
```

Comment:
It would be good to structure the layout of source code files that implement a specific memory-code in some way. For example, `database/engine/{dbengine,ram,mongodb}/*.{c,h}`, or any other scheme you might prefer.

Reply:
Currently, the `database/engine` directory is specific to `dbengine`.

So following your suggestion while minimizing changes, I put the ram-based engine (mmap, ram etc.) code in `database/ram`.